### PR TITLE
AUT-3861: Add new lambda for the /token endpoint.

### PR DESCRIPTION
**Pull request by andrew-moores**
Initial version of the lambda will only handle POST requests and will always return a success response.

### CODE REVIEW

File: ipv-stub/template.yaml [hunk 1]
```diff
@@ -101,9 +117,9 @@ Resources:
                         type: object
               x-amazon-apigateway-integration:
                 type: "aws_proxy"
-                httpMethod: "GET"
+                httpMethod: "POST"
```
Comment (mattvot) [inline ipv-stub/template.yaml:120]: /reverification API here is implemented as GET atm

Comment (andrew-moores) [reply ipv-stub/template.yaml:120]: This caught me out as well.  The integration is always done as a POST regardless of the originating method.

File: ipv-stub/ipv-stub-requests.http [hunk 2]
```diff
@@ -0,0 +1,13 @@
+### AUTHORIZE
+GET https://ipvstub.signin.dev.account.gov.uk/authorize
```
Comment (BeckaL) [inline ipv-stub/ipv-stub-requests.http:2]: Just wondering, what's the value of having this one? It won't work without a generated request - do we want to maintain a hard-coded request in here that works with the current validation and decryption?

File: ipv-stub/template.yaml (hunk 1 above)
Comment (mattvot) [reply ipv-stub/template.yaml:120]: Ah, TIL

File: ipv-stub/ipv-stub-requests.http (hunk 2 above)
Comment (mattvot) [reply ipv-stub/ipv-stub-requests.http:2]: May be possible to import the script. I see IDEA has an example

```
### Composing scripts with imports
< {%
    // common script parts can be moved to separate files and imported with 'import' keyword
    import {makeSignature} from "./my-utils";

    request.variables.set("signature", makeSignature())
%}
POST https://examples.http-client.intellij.net/post
X-My-Signature: {{signature}}
Content-Type: application/json
```

File: ipv-stub/src/endpoints/ipv-token.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    const accessToken = base64url.encode(randomBytes(32));
+    await putReverificationWithAccessToken(accessToken, reverification);
+  } else {
+    logger.info("Did not find reverification result record");
+  }
+  
+  // If exists return an access token.
+
+  return successfulJsonResult(200, {
+    access_token: 'access-token',
```
Comment (andrew-moores) [inline ipv-stub/src/endpoints/ipv-token.ts]: Fixed to return accessToken generated above in subsequent commit.

File: ipv-stub/src/helper/result-helper.ts [hunk 4]
```diff
@@ -88,3 +88,15 @@ export async function handleErrors(
     };
   }
 }
+
+export function shouldObfuscate(paramName: string): boolean {
+  return ["code", "jti", "client_assertion"].includes(paramName);
+}
+
+export function obfuscate(value: string): string {
```
[the remark this answers is not in the document]
Comment (andrew-moores) [reply ipv-stub/src/helper/result-helper.ts:96]: Yes, will fix on a future bau change.

File: ipv-stub/src/endpoints/ipv-token.ts [hunk 5]
```diff
@@ -16,11 +18,124 @@ import {
...
+import { PutCommandOutput } from "@aws-sdk/lib-dynamodb";
 
+const public_key = `
+-----BEGIN PUBLIC KEY-----
+MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEDn0sV329oTHdahzIuUSWS2xw5GVE
+IKUQ9FPvvEDsNKofkw3n7hy1orQQ0XucyhLAcJy0mofJ3fwbjIZEgKBfUw==
+-----END PUBLIC KEY-----
+`.trim();
+
+type Result<T> =
```
[the remark this answers is not in the document]
Comment (andrew-moores) [reply ipv-stub/src/endpoints/ipv-token.ts:30]: Yes, will rationalise our keys when we have heard back from Richard about the questions Aidan had on the keys.

File: ipv-stub/src/endpoints/ipv-token.ts [hunk 6]
```diff
@@ -16,11 +18,124 @@ import {
...
+  }
+
+  logger.info("Handling token request with parameters:");
+  for (const [key, value] of Object.entries(validParameters)) {
+    const loggedValue = shouldObfuscate(key) ? obfuscate(value) : value;
+    logger.info(`${key}::${loggedValue}`);
+  }
+
+  if (missingParamaters.length > 0) {
+    const missingParametersErrorMessgae = `Missing or empty parameters: ${missingParamaters.join(", ")}`;
```
[the remark this answers is not in the document]
Comment (andrew-moores) [reply ipv-stub/src/endpoints/ipv-token.ts:74]: Will pick this up as bau as well.

**Pull request merged**